hazelcast/src/main/java/com/hazelcast/instance/Node.java

```diff
@@ -695,6 +704,7 @@ public void setAsMaster() {
         setJoined();
```

Comment:
I think `setJoined()` should be the last statement, once node is observed as `joined`, cluster version (and other props) should be visible.